lib/pa_ess/utilities.ex

```diff
@@ -509,6 +510,46 @@ defmodule PaEss.Utilities do
 
   def prediction_route_name(%{route_id: route_id}), do: route_id
 
+  defp prediction_seconds(prediction, true) do
+    prediction.seconds_until_departure
+  end
+
+  defp prediction_seconds(prediction, false) do
+    prediction.seconds_until_arrival || prediction.seconds_until_departure
+  end
+
+  @spec prediction_minutes(Predictions.Prediction.t(), boolean()) :: {integer(), boolean()}
+  def prediction_minutes(prediction, terminal?) do
+    sec = prediction_seconds(prediction, terminal?)
+    min = round(sec / 60)
+
+    cond do
+      prediction.stopped_at_predicted_stop? and (!terminal? or sec <= 60) -> {:boarding, false}
+      !terminal? and sec <= 30 -> {:arriving, false}
+      min > 60 -> {60, true}
+      prediction.type == :reverse and min > 20 -> {div(min, 10) * 10, true}
+      true -> {max(min, 1), false}
+    end
+  end
+
+  @spec prediction_approaching?(Predictions.Prediction.t(), boolean()) :: boolean()
+  def prediction_approaching?(prediction, terminal?) do
+    !terminal? and !prediction.stopped_at_predicted_stop? and
+      prediction_seconds(prediction, terminal?) in 31..60
```

Comment:
Technically `terminal?` can only ever be `false` when `predictions_seconds/2` is called here. Should we just explicitly pass `false` as the second param?

Reply:
I kind of went back and forth on that. Technically we could just hardcode it, but it creates a little bit of unnecessary coupling in the logic, so I'm inclined to leave it this way.